File: kjerneinfo/kjerneinfo-consumer/src/main/java/no/nav/kjerneinfo/consumer/organisasjon/OrganisasjonConsumerConfig.java
package no.nav.kjerneinfo.consumer.organisasjon;

import no.nav.common.health.HealthCheckResult;
import no.nav.common.health.selftest.SelfTestCheck;
import no.nav.common.sts.SystemUserTokenProvider;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class OrganisasjonConsumerConfig {

    private static final String ORG_NUMMER_NAV = "990983666";
    private static final String ORG_NAVN_NAV = "NAV IKT";


    @Autowired
    private SystemUserTokenProvider systemUserTokenProvider;

    @Bean
    public OrganisasjonService organisasjonService() {
        return new OrganisasjonServiceImpl(organisasjonV1RestClient());
    }

    @Bean
    public OrganisasjonV1RestClient organisasjonV1RestClient() {
        return new OrganisasjonRestClientImpl(systemUserTokenProvider);
    }

    @Bean
    public SelfTestCheck eregOrganisasjonCheck() {
        return new SelfTestCheck(
                String.format("EREG Organisasjon  %s", ORG_NUMMER_NAV),
                false,
                () -> {
                    try {
                        OrganisasjonResponse organisasjonResponse = organisasjonV1RestClient().hentKjernInfoFraRestClient(ORG_NUMMER_NAV);
                        if (organisasjonResponse.getNavn().getNavnelinje1().equals(ORG_NAVN_NAV))
                            return HealthCheckResult.healthy();
                        else
                            return HealthCheckResult.unhealthy("Feil ved henting av data. Sjekke tjeneste logg");
                    } catch (Exception e) {
                        return HealthCheckResult.unhealthy(e);
                    }
                }
        );
    }
}

File: kjerneinfo/kjerneinfo-consumer/src/main/java/no/nav/kjerneinfo/consumer/organisasjon/OrganisasjonConsumerConfig.kt
package no.nav.kjerneinfo.consumer.organisasjon

import no.nav.common.health.HealthCheckResult
import no.nav.common.health.selftest.SelfTestCheck
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
open class OrganisasjonConsumerConfig {

    @Bean
    open fun organisasjonV1Client(): OrganisasjonV1Client {
        return OrganisasjonV1ClientImpl()
    }

    @Bean
    open fun organisasjonService(client: OrganisasjonV1Client): OrganisasjonService {
        return OrganisasjonServiceImpl(client)
    }

    @Bean
    open fun eregOrganisasjonCheck(): SelfTestCheck {
        return SelfTestCheck(
            String.format("EREG Organisasjon  %s", ORG_NUMMER_NAV),
            false
        ) {
            try {
                val (_, navn) = requireNotNull(organisasjonV1Client().hentNokkelInfo(ORG_NUMMER_NAV))
                if (navn.navnelinje1 == ORG_NAVN_NAV)
                    HealthCheckResult.healthy()
                else
                    HealthCheckResult.unhealthy("Feil ved henting av orgnavn fra Ereg. Sjekke tjeneste logg")
            } catch (e: Exception) {
                HealthCheckResult.unhealthy(e)
            }
        }
    }

    companion object {
        private const val ORG_NUMMER_NAV = "990983666"
        private const val ORG_NAVN_NAV = "NAV IKT"
    }
}
